tests/jni_api.rs

```diff
@@ -371,6 +371,30 @@ pub fn new_primitive_array_wrong() {
     assert_pending_java_exception(&env);
 }
 
+#[test]
+fn get_super_class_ok() {
+    let env = attach_current_thread();
+    let result = env.get_superclass(ARRAYLIST_CLASS);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+}
+
+#[test]
+fn get_super_class_null() {
+    let env = attach_current_thread();
+    let class = env.get_superclass("java/lang/Object").unwrap();
+    assert!(class.is_null());
+}
+
+#[test]
+fn get_super_class_wrong() {
+    let env = attach_current_thread();
+    // this triggers the "java.lang.ClassNotFoundException" (even if spec says it doesn't throw)
```

Comment:
The thing is — the method described in the spec does not take a string, but a reference to a class instance. Here a lookup of a class fails, not `get_super_class`.

Reply:
Thanks, corrected.